docs/diagrams/ParseCommandSequence.puml

```diff
@@ -0,0 +1,48 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+participant ":Game" as Game
+participant ":Parser" as Parser
+participant ":Command" as Command
+participant ":CommandResult" as CommandResult
+participant "coord:Coordinate" as Coordinate
+participant "board:ChessBoard" as ChessBoard
+participant "promoteFrom: ChessPiece" as ChessPiece
+
+Game -> Parser: parseCommand
+activate Parser
+create Command
+Parser -> Command ++
+Command --> Parser --: Command
+Game <-- Parser --: Command
+
+Game -> Command ++: execute
+alt MoveCommand
+    Command -> Parser ++: parseMove
+    Parser --> Command --: Move
+else ShowMoveCommand
+    create Coordinate
+    Command -> Coordinate ++: parseAlgebraicCoor
+    Coordinate --> Command --: coord
+    Command -> ChessBoard ++: showAvailableCoordinates
+    ChessBoard --
+    Command -> ChessBoard ++: getAvailableCoordinatesString
+    ChessBoard --> Command --: String
```

Comment:
Perhaps using coordinateString: String is clearer?
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/82555990/c6f354b6-f0f8-4ab2-b76d-aa268088f175)